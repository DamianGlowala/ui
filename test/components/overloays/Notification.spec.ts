import { describe, it, expect } from 'vitest'
import { UNotification } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'
import ui from '../../../src/runtime/ui.config/overlays/notification'

describe('Notification', () => {
  it.each([
    ['basic case', { props: { ui: { default: { closeButton: ui.default.closeButton } } }}],
    // ['with title', { props: { title: 'Notification title', ui: { default: { closeButton: ui.default.closeButton } } }}],
    // ['with description', { props: { description: 'Notification description', ui: { default: { closeButton: ui.default.closeButton } } }}],
    // ['with icon', { props: { icon: 'i-heroicons-academic-cap', ui: { default: { closeButton: ui.default.closeButton } } }}],
    // ['with avatar', { props: { avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' }, ui: { default: { closeButton: ui.default.closeButton }, avatar: { size: ui.avatar.size } } }}],
    // ['with close button', { props: { closeButton: { label: 'close', color: 'red' }, ui: { default: { closeButton: ui.default.closeButton } } }}],
    // ['with color', { props: {color:'red', ui: { default: { closeButton: ui.default.closeButton } } }}],
    // // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UNotification.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UNotification)
    expect(html).toMatchSnapshot()
  })
})
