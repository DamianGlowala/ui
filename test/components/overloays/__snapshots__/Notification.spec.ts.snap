// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification > renders basic case correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="">
          <p class=""></p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100.02%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with avatar correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center">
        <!--v-if--><span class="relative inline-flex items-center justify-center flex-shrink-0 rounded-full h-10 w-10 text-base"><img class="rounded-full h-10 w-10 text-base" alt="nuxtui" src="https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013"><!--v-if--></span>
        <div class="">
          <p class=""></p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with close button correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="">
          <p class=""></p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-red-500 hover:text-red-600 disabled:text-red-500 dark:text-red-400 dark:hover:text-red-500 dark:disabled:text-red-400 underline-offset-4 hover:underline focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-red-500 dark:focus-visible:ring-red-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span><span class="">close</span>
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with color correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="">
          <p class=""></p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-red-500 dark:bg-red-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with description correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-start">
        <!--v-if-->
        <!--v-if-->
        <div class="">
          <p class=""></p>
          <p class="">Notification description</p>
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with icon correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center"><span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5 text-primary-500 dark:text-primary-400"></span>
        <!--v-if-->
        <div class="">
          <p class=""></p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with title correctly 1`] = `
"<transition-stub appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="">
      <div class="flex items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="">
          <p class="">Notification title</p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" style="width: 100%;"></div>
    </div>
  </div>
</transition-stub>"
`;
